numba/dispatcher.py

```diff
@@ -607,6 +620,8 @@ def typeof_pyval(self, val):
         else:
             if tp is None:
                 tp = types.pyobject
+        # attempt to convert List
+        tp = self._convert_reflected_list(tp)
```

Comment:
While this means `numba.typeof` will produce a different result, I don't know if there's a better way. This might have to do it for now.

Reply:
We could maybe check the module where `typeof` is called for the `__future__.disable_typed_list` import.